contribs/gmf/src/datasource/Manager.js

```diff
@@ -685,88 +724,119 @@ const exports = class {
+    const layersParam = params['LAYERS'];
+    const layersList = layersParam.split(',');
+    googAsserts.assert(layersList.length >= 1);
 
     const filterParam = 'FILTER';
-    let filterParamValue = null;
-
-    if (filterString) {
-      const params = source.getParams();
-      const layersParam = params['LAYERS'];
-      const layersList = layersParam.split(',');
-      googAsserts.assert(layersList.length >= 1);
-
-      if (layersList.length === 1) {
-        // When there's only one layer in the `LAYERS` parameters, then
-        // the filter string is given as-is.
-        filterParamValue = filterString;
-      } else {
-        // When there's more then one layer, then each filter must be wrapped
-        // between parenthesis and the order must also match the `LAYERS`
-        // parameter as well.
-        const filterParamValues = [];
-        for (let i = 0, ii = layersList.length; i < ii; i++) {
-          if (layersList[i] === filtrableLayerName) {
-            filterParamValues.push(`(${filterString})`);
-          } else {
-            filterParamValues.push('()');
+    const filterParamValues = [];
+    let hasFilter = false;
+    for (let i = 0, ii = layersList.length; i < ii; i++) {
+      let filterParamValue = '()';
+
+      const dataSources = this.dataSources_.getArray();
+      for (const dataSource of dataSources) {
+        const dsLayer = this.getDataSourceLayer_(dataSource);
+        if (dsLayer == undefined) {
+          continue;
+        }
+        if (olBase.getUid(dsLayer) == olBase.getUid(layer) &&
+            layersList[i] === dataSource.name)  {
+
+          const id = olBase.getUid(dataSource.gmfLayer);
+          const item = this.treeCtrlCache_[id];
+          googAsserts.assert(item);
+          const treeCtrl = item.treeCtrl;
+          const projCode = treeCtrl.map.getView().getProjection().getCode();
+
+          const filterString = dataSource.visible ?
+            this.ngeoRuleHelper_.createFilterString({
+              dataSource: dataSource,
+              projCode: projCode,
+              incDimensions: true
+            }) :
+            null;
+          if (filterString) {
+            filterParamValue = `(${filterString})`;
+            hasFilter = true;
           }
         }
-        filterParamValue = filterParamValues.join('');
       }
+
+      filterParamValues.push(filterParamValue);
     }
 
     source.updateParams({
-      [filterParam]: filterParamValue
+      [filterParam]: hasFilter ? filterParamValues.join('') : null
     });
   }
 
+  /**
+   * Called when both the 'visible' and 'filterRules' properties of a data
+   * source change.
+   *
+   * If the data source is filtrable, then make sure that when it gets rules
+   * set to apply them as OGC filters to the OpenLayers layer, more precisely
+   * as a `FILTER` parameter in the layer's source parameters.
+   *
+   * @param {!gmf.datasource.OGC} dataSource Data source.
+   * @param {boolean} value Value.
```

Comment:
Value ? What is it ? It is used ?

Reply:
You're right, value is not used, it was already the case before my commit. I will remove it.